src/Sulu/Bundle/AdminBundle/Tests/Unit/FormMetadata/Validation/BlockFieldMetadataValidatorTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Tests\Unit\FormMetadata\Validation;
+
+use PHPUnit\Framework\TestCase;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FieldMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FormMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\ItemMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\SectionMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\Validation\BlockFieldMetadataValidator;
+use Sulu\Component\Content\Exception\ReservedPropertyNameException;
+
+class BlockFieldMetadataValidatorTest extends TestCase
+{
+    /**
+     * @var BlockFieldMetadataValidator
+     */
+    private $blockFieldMetadataValidator;
+
+    protected function setUp(): void
+    {
+        $this->blockFieldMetadataValidator = new BlockFieldMetadataValidator();
+    }
+
+    /**
+     * @param FormMetadata[] $types
+     */
+    private function createFieldMetadata(
+        string $name,
+        string $type = 'block',
```

Comment:
Same as above. But if you prefer we could also introduce another method like `createBlockFieldMetadata` I would also be fine with that. Then a situation like mentioned above, when you don't really know what's happening from reading the code, cannot happen.